modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -48,5 +58,58 @@ object ConstructionService {
 
       NonEmptyList.fromList(accountIdentifiers)
     }
-  }
+
+    def transactionToOperations(transaction: Transaction): Either[String, NonEmptyList[Operation]] = {
+      refineV[Not[Equal[0L]]](transaction.amount.value.value).flatMap { amount =>
+        val operations: Either[String, List[Operation]] = refineV[Not[Equal[0L]]](-1L * transaction.amount.value.value).map { negAmount =>
+            val negativeOperation = Operation(
+              OperationIdentifier(OperationIndex(0L)),
+              none,
+              OperationType.Transfer,
+              none,
+              AccountIdentifier(transaction.destination, none).some,
+              Amount(AmountValue(negAmount), DAG).some
+            )
+
+            val positiveOperation = Operation(
+              OperationIdentifier(OperationIndex(1L)),
+              none,
+              OperationType.Transfer,
+              none,
+              AccountIdentifier(transaction.destination, none).some,
+              Amount(AmountValue(amount), DAG).some
+            )
+
+            List(negativeOperation, positiveOperation)
+        }
+
+        operations.flatMap{ operationList =>
+          NonEmptyList.fromList(operationList) match {
+            case Some(value) => value.asRight
+            case None => "No valid operations found.".asLeft
+          }
+        }
+      }
+    }
+
+  def parseSignedTransaction(hex: Hex): F[Either[MalformedTransaction.type, ConstructionParse.Response]] = 
+    KryoSerializer[F].deserialize[Signed[Transaction]](hex.toBytes).traverse { signedTransaction =>
+    transactionToOperations(signedTransaction).traverse { operationsList =>
+      signedTransaction.proofs.toNonEmptyList.traverse(signature => signature.id.hex.toPublicKey).map { proofs =>
+        val proofAccountIdentifiers = proofs.map(_.toAddress).map { address =>
+          AccountIdentifier(address, none)
+        }
+        
+        ConstructionParse.Response(operationsList.toList, proofAccountIdentifiers.toList.some)
+      }
+    }.map(_.left.map(_ => MalformedTransaction))
+  }.map(_.left.map(_ => MalformedTransaction).flatten)
+
+  def parseUnsignedTransaction(hex: Hex): Either[MalformedTransaction.type, ConstructionParse.Response] = 
```

Comment:
Here you have `MalformedTransaction.type` but in trait you have `ParseFailure.type`. Which one is valid? Because it won't compile because it wants `ParseFailure` but you pass `MalformedTransaction`

Reply:
@marcinwadon requested reusing `MalformedTransaction`. I forgot to update the trait. That has been rectified.